web/Services/ApiService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using BricksAndHearts.Auth;
+using BricksAndHearts.ViewModels;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace BricksAndHearts.Services;
+
+public interface IApiService
+{
+    public Task<string> MakeApiRequestToAzureMaps(string postalCode);
+    public Task<PropertyViewModel> AutofillAddress(PropertyViewModel model);
+    public PostcodeApiResponseViewModel TurnResponseBodyToModel(string responseBody);
+}
+
+public class ApiService : IApiService
+{
+    private readonly HttpClient _client = new HttpClient();
+    private readonly ILogger<ApiService> _logger;
+    private readonly IOptions<AzureMapsOptions> _options;
+
+    public ApiService(ILogger<ApiService> logger, IOptions<AzureMapsOptions> options)
+    {
+        _logger = logger;
+        _options = options;
+    }
+
+    public async Task<string> MakeApiRequestToAzureMaps(string postalCode)
```

Comment:
This feels like it should be private, and is just public for testing. That isn't always wrong, but usually means there's a better way to organise things. For example we could have this function just deal with talking to Azure and have the autofilling logic somewhere else - either a new file or in PropertyService. (no need to change it, just for next time)

Reply:
ok I see :-) thank you!